src/CMakeLists.txt

```diff
@@ -16,14 +16,17 @@ set(CUDA_SOURCES
     memory/fill.cu
 )
 
-if(${WITH_MPI})
+if(NMC_WITH_MPI)
     set(BASE_SOURCES ${BASE_SOURCES} communication/mpi.cpp)
 endif()
 
 add_library(nestmc ${BASE_SOURCES} ${HEADERS})
 
-add_dependencies(nestmc build_all_mods)
-if(WITH_CUDA)
+if (NMC_AUTO_RUN_MODCC_ON_CHANGES)
```

Comment:
Well actually thinking about it, the default value for `AUTO_GENERATE` is ON, so on first checkout, the mod files will be created. Only when an 'advanced' user disables it (presumably on purpose) will the generation be turned off, so it should not be a problem.